viewing_party/party.py

```diff
@@ -1,23 +1,212 @@
 # ------------- WAVE 1 --------------------
 
+from enum import unique
+from itertools import count
+
+# creates a movie object with a given title, genre, and rating
+# parameters: title - title of movie, genre - genre of movie, rating - rating of the movie 
 def create_movie(title, genre, rating):
-    pass
+    # we first check if all of our arguments are truthy if not we return None
+    if not title or not genre or not rating:
+        return None
+    # we then create the dictionary with the keys title, genre, and rating corresponding to our given arguments and return
+    else:
+        movie_details = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+        return movie_details
+
+# adds a movie object to a user's watched list
+# parameters: user_data - a user's movie information including watched list, movie - movie you have watched
+def add_to_watched(user_data, movie):
+    # we can append the movie object to the user's watched list
+    user_data["watched"].append(movie)
+    return user_data
+
+# adds a movie object to a user's want to watch list
+# parameters: user_data - a user's movie information including watched list, movie - movie you want to watch
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
 
+# moves a movie object from a user's want to watch list to their watched list, different from add_to_watched() which adds a movie object not already in the want to watch list to the watch list
+def watch_movie(user_data, title):
+    # look for the movie in the want to watch list by looping through and matching the titles
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            # using the add_to_watched() function add the movie object to the watched list 
+            add_to_watched(user_data, movie)
+            # we can now remove the movie from the user's want to watch list
+            user_data["watchlist"].remove(movie)
+            # we return here and end the loop
+            return user_data
+    # the function will not add the move to the watched list if it was not found in the want to watch list
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+# calculates a user's average rating for movies from their watched list (let's see how their taste is)
+# parameters: user_data - a user's movie information including watched list
+# returns float average rating, 0.0 if empty watched list
+def get_watched_avg_rating(user_data):
+    # set the avg rating to 0 initially, if they have no movies in their watched list this is the default value returned
+    average_rating = 0.0

```

Comment:
Minor naming nitpick: Despite being called `average_rating`, this actually starts out getting filled with the sum of ratings before it gets reassigned on line 62 below. It would perhaps be a bit clearer to name this `sum_of_ratings`.